### PR TITLE
[QSP-9, QSP-20] add bounds to optimisticSeconds

### PR DESCRIPTION
**Pull request by anna-carroll**
## Motivation
We wish to add some reasonable input validation to the optimisticSeconds field

## Solution
Ensure optimisticSeconds is greater than 25 minutes & less than 1 year. 

## PR Checklist

- [ ] Added Tests
- [ ] Updated Documentation
- [ ] Updated CHANGELOG.md for the appropriate package

### CODE REVIEW

**Comment by anna-carroll** [comment]
realized while making this PR that, ofc, we set `optimisticTimeout` to 10 seconds in our development testnet environment 😭

**Comment by odyslam** [comment]
Although I am generally against input validation for gas optimization reasons, I think this codepth is rare enough to allow for input validation

**Comment by anna-carroll** [comment]
> rare enough to allow for input validation

yes, but for our development testnet environment, we want to use a value for `optimisticSeconds` which is not within the bounds of reason for production (10 seconds for dev environment)

File: packages/contracts-core/contracts/Replica.sol [hunk 1]
```diff
@@ -345,6 +343,22 @@ contract Replica is Version0, NomadBase {
...
     // ============ Internal Functions ============
 
+    /**
+     * @notice Set optimistic timeout period for new roots
+     * @dev Called by owner (Governance) or at initialization
+     * @param _optimisticSeconds New optimistic timeout period
+     */
+    function _setOptimisticTimeout(uint256 _optimisticSeconds) internal {
+        // ensure the optimistic timeout is at least 25 minutes
+        require(_optimisticSeconds >= 1500, "optimistic timeout too low");
```
Comment (prestwich) [inline packages/contracts-core/contracts/Replica.sol:353]: what about testing environments?

**Comment by anna-carroll** [comment]
we don't want to address this for test environments

**Comment by prestwich** [comment]
We want to address this, but with no minimum bound

File: packages/contracts-core/contracts/Replica.sol [hunk 2]
```diff
@@ -359,6 +357,25 @@ contract Replica is Version0, NomadBase {
...
+     * @notice Set optimistic timeout period for new roots
+     * @dev Called by owner (Governance) or at initialization
+     * @param _optimisticSeconds New optimistic timeout period
+     */
+    function _setOptimisticTimeout(uint256 _optimisticSeconds) internal {
+        // This allows us to initialize the value to be very low in test envs,
+        // but does not allow governance action to lower a production env below
+        // the safe value
+        uint256 _current = optimisticSeconds;
+        if (_current != 0 && _current > 1500) require(_optimisticSeconds >= 1500, "optimistic timeout too low");
```
Comment (prestwich) [inline packages/contracts-core/contracts/Replica.sol:370]: @anna-carroll please review this line carefully

**Review by prestwich — APPROVED** [review]
approved, provided you thoroughly check my if/require

File: packages/contracts-core/contracts/Replica.sol (hunk 2 above)
Comment (anna-carroll) [reply packages/contracts-core/contracts/Replica.sol:370]: hmm prevents us from updating `optimisticSeconds` in test envs. should be acceptable tho.. let me think a sec

Comment (prestwich) [reply packages/contracts-core/contracts/Replica.sol:370]: it shouldn't prevent us from updating in test envs. It should be a ratchet. I.e. once above 1500, never below 1509

Comment (anna-carroll) [reply packages/contracts-core/contracts/Replica.sol:370]: what if 

if current is zero, allow anything 
if current is less than 1500, require it's greater than current
if current is geq 1500, require it's geq 1500

Comment (anna-carroll) [reply packages/contracts-core/contracts/Replica.sol:370]: yep, okay. good

**Review by anna-carroll — COMMENTED** [review]
good.

**Comment by anna-carroll** [comment]
if current = 0 (never set before), allow anything
if current < 1500 (testnet environments), allow anything
if current > 1500 (prod), require GEQ 1500

**Pull request merged**